src/fsrs/fsrs.py

```diff
@@ -58,9 +58,18 @@ def repeat(
             s.easy.scheduled_days = easy_interval
             s.easy.due = now + timedelta(days=easy_interval)
         elif card.state == State.Learning or card.state == State.Relearning:
+            short_term_stability_good = self.short_term_stability(
+                card.stability, Rating.Good
+            )
+            short_term_stability_easy = self.short_term_stability(
+                card.stability, Rating.Easy
+            )
+
```

Comment:
Could it be due to not updating the difficulty? See here for more details: https://github.com/open-spaced-repetition/ts-fsrs/blob/65fd676414e23e21612b5344af947480f7dafa7e/src/fsrs/impl/basic_schduler.ts#L68-L111.

Reply:
I went and added
```python
  s.again.difficulty = self.next_difficulty(card.difficulty, Rating.Again)
  s.hard.difficulty = self.next_difficulty(card.difficulty, Rating.Hard)
  s.good.difficulty = self.next_difficulty(card.difficulty, Rating.Good)
  s.easy.difficulty = self.next_difficulty(card.difficulty, Rating.Easy)
```
just below and now when I set `w17=0 and w18=0`, we get the same results. So I think this means that the rest of the issue is left in one of our short term stability functions (probably mine)

Reply:
Okay I think I got it! I'll send some commits in a bit